perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -1290,6 +1330,11 @@ public static boolean isAuthorizedForAttribute(PerunSession sess, AttributeActio
 		log.trace("Entering isAuthorizedForAttribute: sess='{}', actionType='{}', attrDef='{}', primaryHolder='{}', " +
 			"secondaryHolder='{}'", sess, actionType, attrDef, resource, null);
 
+		// Check roles which are authorized by default
+		if (sess.getPerunPrincipal().getRoles() != null && !sess.getPerunPrincipal().getRoles().isEmpty() && hasAccessByDefault(sess, actionType)) {
```

Comment:
Just a nit: maybe we could move the first two conditions to hasAccessByDefault so it is more readable.

Reply:
Good point, moved it. Also added test for the write case.